src/sentry/sentry_metrics/querying/metadata.py

```diff
@@ -0,0 +1,185 @@
+from datetime import datetime
+from typing import Generator, List, Optional, Sequence, Set, TypedDict
+
+from sentry.api.utils import InvalidParams
+from sentry.models.organization import Organization
+from sentry.models.project import Project
+from sentry.sentry_metrics.querying.utils import get_redis_client_for_ingest
+from sentry.utils import json, metrics
+
+DAY_IN_SECONDS = 86400
+
+
+class CodeLocationQuery(TypedDict):
+    organization_id: int
+    project_id: int
+    metric_mri: str
+    timestamp: int
+
+
+class CodeLocationPayload(TypedDict):
+    function: Optional[str]
+    module: Optional[str]
+    filename: Optional[str]
+    abs_path: Optional[str]
+    lineno: Optional[int]
+
+
+class MetricCodeLocations(TypedDict):
+    query: CodeLocationQuery
+    code_locations: Sequence[CodeLocationPayload]
```

Comment:
can we keep fields camelCased in the responses we are returning. We tend to do it on most of the other endpoints

Reply:
I can do it but I will cry while doing that, I will try my best to see how I can use the serializers for that.